extensions/VisualEditor/wikia/modules/ve/dm/ve.dm.WikiaTransclusionModel.js

```diff
@@ -2,178 +2,180 @@
+					specs[page.title].params[page.params[i]] = {};
+				}
 			}
+			ve.extendObject( this.specCache, specs );
 		}
-		ve.extendObject( this.specCache, specs );
-	}
-};
-
-/**
- * @param {Object} specs
- * @param {Object} node
- * @param {string} sourceName
- * @param {Object} sourceMetadata
- */
-function addInfoboxSourceParam( specs, node, sourceName, sourceMetadata ) {
-	specs.params[sourceName] = {
-		label: sourceMetadata.label
 	};
 
-	if ( sourceMetadata.primary === true ) {
-		specs.params[sourceName].type = node.type;
-	}
+	/**
+	 * @param {Object} specs
+	 * @param {Object} node
+	 * @param {string} sourceName
+	 * @param {Object} sourceMetadata
+	 */
+	function addInfoboxSourceParam( specs, node, sourceName, sourceMetadata ) {
+		specs.params[sourceName] = {
+			label: sourceMetadata.label
+		};
+
+		if ( sourceMetadata.primary === true ) {
+			specs.params[sourceName].type = node.type;
+		}
 
-	specs.paramOrder.push( sourceName );
-}
+		specs.paramOrder.push( sourceName );
+	}
 
-/**
- * @param {Object} node
- * @param {Object} specs
- */
-function parseInfoboxMetadata( node, specs ) {
-	if ( node.type === 'group' ) {
-		node.metadata.forEach( function ( childNode ) {
-			parseInfoboxMetadata( childNode, specs );
-		} );
-	} else {
-		Object.keys( node.sources ).map( function ( sourceName ) {
-			var sourceMetadata = node.sources[sourceName];
-			addInfoboxSourceParam( specs, node, sourceName, sourceMetadata );
-		} );
+	/**
+	 * @param {Object} node
+	 * @param {Object} specs
+	 */
+	function parseInfoboxMetadata( node, specs ) {
+		if ( node.type === 'group' ) {
+			node.metadata.forEach( function ( childNode ) {
+				parseInfoboxMetadata( childNode, specs );
+			} );
+		} else {
+			Object.keys( node.sources ).map( function ( sourceName ) {
+				var sourceMetadata = node.sources[sourceName];
+				addInfoboxSourceParam( specs, node, sourceName, sourceMetadata );
+			} );
+		}
 	}
-}
 
-/**
- * @desc process the infobox params received from API. For all requested pages, if they contain infoboxes,
- * extend their param array with received infobox params.
- */
-ve.dm.WikiaTransclusionModel.prototype.fetchInfoboxParamsRequestDone = function ( specs, data ) {
-	var id, page, denormalizedData;
+	/**
+	 * @desc process the infobox params received from API. For all requested pages, if they contain infoboxes,
+	 * extend their param array with received infobox params.
+	 */
+	ve.dm.WikiaTransclusionModel.prototype.fetchInfoboxParamsRequestDone = function ( specs, data ) {
+		var id, page, denormalizedData;
 
-	if ( data && Object.keys( data.query ).length && Object.keys( data.query.pages ).length ) {
-		denormalizedData = this.denormalizeInfoboxTemplateTitles( data );
+		if ( data && Object.keys( data.query ).length && Object.keys( data.query.pages ).length ) {
+			denormalizedData = this.denormalizeInfoboxTemplateTitles( data );
 
-		for ( id in denormalizedData ) {
-			page = denormalizedData[id];
+			for ( id in denormalizedData ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Restrict what this loop acts on by testing each property. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)